contracts/strategies/direct-grants-simple/DirectGrantsSimpleStrategy.sol

```diff
@@ -166,38 +169,42 @@ contract DirectGrantsSimpleStrategy is BaseStrategy, ReentrancyGuard {
     /// @notice Set milestones for recipient
     /// @param _recipientId Id of the recipient
     /// @param _milestones The milestones to be set
-    function setMilestones(address _recipientId, Milestone[] memory _milestones) external onlyPoolManager(msg.sender) {
-        if (upcomingMilestone[_recipientId] != 0) {
-            revert MILESTONES_ALREADY_SET();
-        }
-
-        Recipient memory recipient = _getRecipient(_recipientId);
+    function setMilestones(address _recipientId, Milestone[] memory _milestones) external {
+        Recipient storage recipient = _recipients[_recipientId];
 
         if (recipient.recipientStatus != InternalRecipientStatus.Accepted) {
             revert RECIPIENT_NOT_ACCEPTED();
         }
 
-        uint256 totalAmountPercentage;
-
-        uint256 milestonesLength = _milestones.length;
-        for (uint256 i = 0; i < milestonesLength;) {
-            Milestone memory milestone = _milestones[i];
-            if (milestone.milestoneStatus != RecipientStatus.None) {
-                revert INVALID_MILESTONE();
-            }
-            totalAmountPercentage += milestone.amountPercentage;
-            milestones[_recipientId].push(milestone);
+        if (recipient.milestonesReviewStatus == RecipientStatus.Accepted) {
+            revert MILESTONES_ALREADY_SET();
+        }
 
-            unchecked {
-                i++;
-            }
+        bool isRecipientCreator = recipient.creator == msg.sender;
```

Comment:
Since the recipientId can only be the sender OR a pool manager, the `creator` property is not needed.
Instead of, we could implement `isRecipientCreator` like that:
`bool isRecipientCreator = ( msg.sender == _recipientId )|| _isProfileMember(recipientId, msg.sender);`